### PR TITLE
Topology clean ups - Part3

### PR DESCRIPTION
**Pull request by ranj063**
Depends on https://github.com/thesofproject/linux/pull/4293

This is a lot of patches but most of them along the same theme to separate the gateways out of the pipelines and add support for building simple copier->copier pipelines for LNL nocodec topology

### CODE REVIEW

**Comment by ranj063** [comment]
SOFCI TEST

**Review by kv2019i — COMMENTED** [review]
This is quite massive set to review. I think all the dai/host-gateway removals look ok, and also removals of deprecated stuff. Not sure about the passthrough nocodec featureu, that seems quite invasive and I wonder whether it would be simpler to duplicate definitions for a passthrough variant. If the include-by-key allows to support more things from one source, but it's like ifdefs in C-code, too much and the code becomes really hard to follow. Not blocking on this still, I do understand the intent and maybe this is best solution.

File: tools/topology/topology2/cavs-rt5682.conf [hunk 1]
```diff
@@ -89,6 +89,7 @@ Define {
 	BT_PB_HOST_PIPELINE_SINK "copier.SSP.10.1"
 	BT_PB_DAI_PIPELINE_SRC "copier.host.9.1"
 	BT_PB_PIPELINE_STREAM_NAME "copier.SSP.10.1"
+	PASSTHROUGH	"false"
```
[the remark this answers is not in the document]
Comment (ranj063) [reply tools/topology/topology2/cavs-rt5682.conf:92]: @kv2019i while adding a new passthrough topology file seems like an easier solution at the outset, it is harder to maintain. There are quite a few details that are difficult to track in terms of the DAI configuration and maintaining this across multiple files will just not be feasible. Taking the existing nocodec tplg and bypassing the processing pipelines is the simplest solution even though it involves a bit of work right now.

[the remark this answers is not in the document]
Comment (plbossart) [reply tools/topology/topology2/cavs-rt5682.conf:92]: I am not so sure @ranj063. 

The drawback is that we end-up pushing conditional parts all the way down in the hope of maintaining a single top-level file. That's what prompted me to just get rid of everything and use the simplest top-level file with just what I needed.

I think we really need to avoid making low-level macros a kitchen sink that is used from a single-place and tons of options. That's not maintainable either.

It's a fine balance to reach indeed.

[the remark this answers is not in the document]
Comment (lgirdwood) [reply tools/topology/topology2/cavs-rt5682.conf:92]: @plbossart @ranj063 @kv2019i any update on alignment here ? Is this better than what we have today ?
